usr/src/uts/armv8/os/fakebop.c

```diff
@@ -479,107 +909,55 @@ read_bootenvrc(void)
 	v_len = do_bsys_getproplen(NULL, "console");
 	if (v_len > 0) {
 		(void) do_bsys_getprop(NULL, "console", consoledev);
+		if (strcmp(consoledev, "graphics") == 0) {
+			bsetprops("console", "text");
+			v_len = strlen("text");
+			bcopy("text", consoledev, v_len);
+		}
 	} else {
 		v_len = 0;
 	}
 	consoledev[v_len] = 0;
-}
-static void
-bsetprop(char *name, int nlen, void *value, int vlen)
-{
-	pnode_t chosen = prom_chosennode();
-	prom_setprop(chosen, name, (const caddr_t)value, vlen);
-}
 
-static void
-bsetprops(char *name, char *value)
-{
-	bsetprop(name, strlen(name), value, strlen(value) + 1);
-}
-
-static void
-bsetprop64(char *name, uint64_t value)
-{
-	bsetprop(name, strlen(name), (void *)&value, sizeof (value));
-}
-
-static void
-bsetpropsi(char *name, int value)
-{
-	char prop_val[32];
-
-	(void) snprintf(prop_val, sizeof (prop_val), "%d", value);
-	bsetprops(name, prop_val);
-}
-
-/*
- * to find the size of the buffer to allocate
- */
-/*ARGSUSED*/
-int
-do_bsys_getproplen(bootops_t *bop, const char *name)
-{
-	pnode_t chosen = prom_chosennode();
-	return prom_getproplen(chosen, name);
-}
+#if XXXARM
+	bcons_post_bootenvrc(inputdev, outputdev, consoledev);
+#endif
 
-int
-do_bsys_getprop(bootops_t *bop, const char *name, void *value)
-{
-	pnode_t chosen = prom_chosennode();
-	prom_getprop(chosen, name, (caddr_t)value);
-	return 0;
+#if XXXARM
+	if (find_boot_prop("prom_debug") || kbm_debug)
+		boot_prop_display(line);
+#endif
 }
 
 /*
- * get the name of the next property in succession from the standalone
+ * print formatted output
  */
-/*ARGSUSED*/
-static char *
-do_bsys_nextprop(bootops_t *bop, char *name)
-{
-	static char next[OBP_MAXPROPNAME];
-	pnode_t chosen = prom_chosennode();
-	return prom_nextprop(chosen, name, next);
-}
-
 void
-vbop_printf(void *ptr, const char *fmt, va_list ap)
+vbop_printf(void *ptr __unused, const char *fmt, va_list ap)
 {
+#if XXXARM
+	if (have_console == 0)
+		return;
+#endif
 	(void) vsnprintf(buffer, BUFFERSIZE, fmt, ap);
-
-	for (int i = 0; i < BUFFERSIZE && buffer[i]; i++) {
-		if (buffer[i] == '\n') {
-			BSVC_PUTCHAR(SYSP, '\r');
-		}
-		BSVC_PUTCHAR(SYSP, buffer[i]);
-	}
+	PUT_STRING(buffer);
 }
 
 void
 bop_printf(void *bop, const char *fmt, ...)
 {
-	va_list	ap;
-	int i;
+	va_list ap;
 
 	va_start(ap, fmt);
-	(void) vsnprintf(buffer, BUFFERSIZE, fmt, ap);
+	vbop_printf(bop, fmt, ap);
 	va_end(ap);
-
-	for (i = 0; i < BUFFERSIZE && buffer[i]; i++) {
-		if (buffer[i] == '\n') {
-			BSVC_PUTCHAR(SYSP, '\r');
-		}
-		BSVC_PUTCHAR(SYSP, buffer[i]);
-	}
 }
 
 
 /*
  * Another panic() variant; this one can be used even earlier during boot than
  * prom_panic().
  */
-/*PRINTFLIKE1*/
```

Comment:
You could add the gcc attribute with `__PRINTFLIKE(1)` in the declaration, so it still error checks.

Reply:
Yeah, another transposition from i86pc - I'll clean it up.